src/services/interactionUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { EventType } from '@vcmap/core'
+import { viewList } from '@/model/views.model'
+import { useViewsStore } from '@/stores/views'
+import { RENNES_LAYER } from '@/stores/layers'
+import SelectRoofInteraction from '@/interactions/selectRoofInteraction'
+import ForbidenClickInteraction from '@/interactions/forbidClickInteraction'
+import type { RennesApp } from '@/services/RennesApp'
+
+function _getSelectRoofInteraction(
+  rennesApp: RennesApp
+): SelectRoofInteraction | undefined {
+  let res = undefined
+  rennesApp.maps.eventHandler.interactions.forEach((interaction) => {
+    if (interaction instanceof SelectRoofInteraction) {
+      res = interaction
+    }
+  })
+  return res
+}
+
+export function enableSelectRoofInteraction(rennesApp: RennesApp) {
+  const res = _getSelectRoofInteraction(rennesApp)
+  if (res) {
+    res.setActive(EventType.CLICKMOVE)
+  }
+}
+
+export function disableSelectRoofInteraction(rennesApp: RennesApp) {
+  const res = _getSelectRoofInteraction(rennesApp)
+  if (res) {
+    res.setActive(false)
+  }
+}
+
+export function cleanHighlightedRoofs(rennesApp: RennesApp) {
+  const res = _getSelectRoofInteraction(rennesApp)
+  if (res) {
+    res.unhighlight()
+  }
+}
+export function createMapInteractions(rennesApp: RennesApp) {
+  const viewStore = useViewsStore()
+  let selectInteraction
+  if (
+    [
+      viewList['roof-selected-information'],
+      viewList['roof-selection'],
+    ].includes(viewStore.currentView)
+  ) {
+    if (_getSelectRoofInteraction(rennesApp) == null) {
+      selectInteraction = new SelectRoofInteraction(
+        rennesApp.maps.layerCollection.getByKey(RENNES_LAYER.roof3d),
+        rennesApp
+      )
+      selectInteraction.setActive(false)
+      rennesApp.maps.eventHandler.addExclusiveInteraction(
+        selectInteraction,
+        () => {}
+      )
+    }
+  } else if (viewStore.currentView === viewList.home) {
+    const forbidClickInteraction = new ForbidenClickInteraction()
```

Comment:
I understand the idea here of separating the part specific to the home page in a new interaction. 
But the problem is that the basic building click interaction is no longer present on the home page. When you click on a building, the checkbox turns red if it is not checked. But if you check it and click on the building, nothing happens. While the user should be sent to the page: /roof-selected-information

Reply:
There was a misunderstanding here. On the home page, when the user clicks on the map if the checkbox is not checked, the error message is displayed. And that's all . The click on the building does not change page.